src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialCapacity) {
+        this.elements = (T[]) new Object[initialCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            elements = grow(size);
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        if (size == elements.length) {
+            elements = grow(size);
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        if (size < elements.length) {
+            elements = grow(size + 1);
+        }
+        System.arraycopy(elements, size, elements, size, size - 1);
```

Comment:
you need this
why do you invoke method grow() when size is less than capacity? you have free space. Why making array bigger
this is absolutely redundant, just remove it